### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by jarrpa**
Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

[skip ci]

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,44 @@
...
+.settings/**
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+
+# Vscode files
+.vscode
+
+# This is where the result of the go build goes
```
Comment (BlaineEXE) [inline .gitignore]: The form of the `_output` directory used by Rook is already in this file below.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,44 @@
...
+[._]*.s[a-w][a-z]
+[._]s[a-w][a-z]
+*.un~
+Session.vim
+.netrwhist
+
+# cscope-related files
+cscope.*
+
+# Mercurial files
```
Comment (BlaineEXE) [inline .gitignore]: Mercurial files are not applicable to Rook.

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,44 @@
...
+**/.hg
+**/.hg*
+
+.tags*
```
Comment (BlaineEXE) [inline .gitignore]: What even is this?

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,44 @@
...
+/output*/
+/_output*/
+/_output
+
+# Emacs save files
+*~
+\#*\#
+.\#*
+
+# Vim-related files
```
Comment (BlaineEXE) [inline .gitignore]: We should either (1) remove this section or (2) include a complete list of vim-related files. https://github.com/github/gitignore/blob/master/Global/Vim.gitignore

I would prefer option 1 without any vim users chiming in

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,44 @@
...
+
+# Vscode files
+.vscode
+
+# This is where the result of the go build goes
+/output*/
+/_output*/
+/_output
+
+# Emacs save files
```
Comment (BlaineEXE) [inline .gitignore]: We should either (1) remove this section or (2) include a complete list of emacs-related files. https://github.com/github/gitignore/blob/master/Global/Emacs.gitignore

I would prefer option 1 without any emacs users chiming in.

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,44 @@
-/.vscode/
+# OSX trash
+.DS_Store
+
+# Eclipse files
+.classpath
+.project
+.settings/**
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
```
Comment (BlaineEXE) [inline .gitignore]: We should either (1) remove this section or (2) include a complete list of jetbrains-related files. https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore

I would prefer option 1 without any jetbrains users chiming in.

File: .gitignore [hunk 7]
```diff
@@ -1,4 +1,44 @@
-/.vscode/
+# OSX trash
+.DS_Store
+
+# Eclipse files
```
Comment (BlaineEXE) [inline .gitignore]: We should either (1) remove this section or (2) include a complete list of eclipse-related files. https://github.com/github/gitignore/blob/master/Global/Eclipse.gitignore

I would prefer option 1 without any eclipse users chiming in.

File: .gitignore [hunk 8]
```diff
@@ -1,4 +1,44 @@
...
+.\#*
+
+# Vim-related files
+[._]*.s[a-w][a-z]
+[._]s[a-w][a-z]
+*.un~
+Session.vim
+.netrwhist
+
+# cscope-related files
```
Comment (BlaineEXE) [inline .gitignore]: Rook does not have C code. This does not apply.

File: .gitignore [hunk 9]
```diff
@@ -1,4 +1,44 @@
-/.vscode/
+# OSX trash
```
Comment (BlaineEXE) [inline .gitignore]: We should either (1) remove this section or (2) include a complete list of macos-related files. 
https://github.com/github/gitignore/blob/master/Global/macOS.gitignore

I would prefer option 1 without any macos users chiming in. @travis / @leseb

**Comment by BlaineEXE** [comment]
Closing this, as nothing is specifically Rook-related. All these can be set by the user in their global `.gitignore` file.

**Comment by jarrpa** [comment]
> Everything not related to Rook should be set locally on your machine by using `git config --global core.excludesfile ~/.gitignore_global`.

Works for me!

**Pull request closed without merging**